src/features/duck-player.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @module Duck Player Overlays
+ *
+ * @description
+ *
+ * Overlays will currently only show on YouTube.com
+ */
+import ContentFeature from '../content-feature.js'
+
+import { DuckPlayerOverlayMessages, OpenInDuckPlayerMsg, Pixel } from './duckplayer/overlay-messages.js'
+import { Messaging, MessagingContext, WindowsMessagingConfig } from '@duckduckgo/messaging'
+import { Environment, initOverlays } from './duckplayer/overlays.js'
+
+/**
+ * @typedef UserValues - A way to communicate user settings
+ * @property {{enabled: {}} | {alwaysAsk:{}} | {disabled:{}}} privatePlayerMode - one of 3 values
+ * @property {boolean} overlayInteracted - always a boolean
+ */
+
+/**
+ * @internal
+ */
+export default class DuckPlayerFeature extends ContentFeature {
+    /**
+     * Lazily create a messaging instance for the given Platform + feature combo
+     * @return {Messaging}
+     */
+    get messaging () {
```

Comment:
As agreed we're going to put this in the `ContentFeature` prototype but it's not worth the bundle size changing for other builds yet.

Please fast follow here too, to clean this up.